ludwig/utils/upload_utils.py

```diff
@@ -205,3 +212,88 @@ def upload(
             return True
 
         return False
+
+
+class Predibase(BaseModelUpload):
+    def __init__(self):
+        self.pc = None
+
+    def login(self):
+        """Login to Predibase using the token stored in the PREDIBASE_API_TOKEN environment variable and returns a
+        PredibaseClient object that can be used to interact with Predibase."""
+        from predibase import PredibaseClient
+
+        try:
+            pc = PredibaseClient()
+
+            # TODO: Check if subscription has expired
+
+            self.pc = pc
+        except Exception as e:
+            raise Exception(f"Failed to login to Predibase: {e}")
+            return False
+
+        return True
+
+    def upload(
+        self,
+        repo_id: str,
+        model_path: str,
+        commit_description: Optional[str] = None,
+        dataset_file: Optional[str] = None,
+        dataset_name: Optional[str] = None,
+    ) -> bool:
+        """Create an empty repo in Predibase and upload trained model artifacts to that repo.
+
+        Args:
+            model_path (`str`):
+                The path of the saved model. This is the top level directory where
+                the models weights as well as other associated training artifacts
+                are saved.
+            repo_name (`str`):
+                A repo name.
+            repo_description (`str` *optional*):
+                The description of the repo.
+        """
+        # Create empty model repo using repo_name, but it is okay if it already exists.
+        try:
+            repo = self.pc.create_repo(
+                name=repo_id,
+                description=commit_description,
+                exist_ok=True,
+            )
+        except Exception as e:
+            raise Exception(f"Failed to create repo in Predibase: {e}")
+            return True
+
+        # Upload the dataset to Predibase
+        try:
+            dataset = self.pc.upload_dataset(
+                file_path=dataset_file,
+                name=dataset_name,
+            )
+        except Exception as e:
+            raise Exception(f"Failed to upload dataset to Predibase: {e}")
+            return True
+
+        with tempfile.TemporaryDirectory() as tmpdir:
```

Comment:
Is this zipping something we can add to the Predibase SDK, like:

```
pc.upload_model_from_dir(...)
```

Or just check in `upload_model` if the path provided is a directory instead of an archive?

Reply:
Good callout! I thought it was funky as well and will update Predibase!